cmd/necogcp/cmd/createrunner.go

```diff
@@ -0,0 +1,195 @@
+package cmd
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"text/template"
+
+	"github.com/cybozu-go/log"
+	"github.com/cybozu-go/neco-gcp/pkg/gcp"
+	"github.com/cybozu-go/neco-gcp/pkg/github"
+	"github.com/cybozu-go/well"
+	"github.com/spf13/cobra"
+	"google.golang.org/api/googleapi"
+)
+
+const (
+	SERVICE_ACCOUNT = "bootstrap-dctest@neco-test.iam.gserviceaccount.com"
+)
+
+var (
+	appId          int64
+	installationId int64
+	privateKeyPath string
+	pat            string
+	runnerRepo     string
+	runnerName     string
+	labels         []string
+)
+
+var runnerStartUpScriptTmpl = template.Must(template.New("").Parse(`#!/bin/sh
+
+useradd -m runner
+usermod -aG google-sudoers runner
+RUNNER_HOME=/home/runner
+/snap/bin/gcloud secrets versions access latest --secret="ghatoken_{{ .RunnerName }}" > $RUNNER_HOME/github_token &&
+cat << EOF >job-cancelled
+#!/bin/sh
+EOF
+chmod 755 job-cancelled
+cp job-cancelled /usr/local/bin/job-started
+cp job-cancelled /usr/local/bin/job-failure
+cp job-cancelled /usr/local/bin/job-success
+mv job-cancelled /usr/local/bin/job-cancelled
+
+NUM_DEVICES=$(ls /dev/nvme0n*|wc -l)
+mdadm --create /dev/md0 -l stripe --raid-devices=${NUM_DEVICES} /dev/nvme0n*
+mkfs -t ext4 -F /dev/md0
+mkdir -p /var/scratch
+mount -t ext4 /dev/md0 /var/scratch
+chmod 1777 /var/scratch
+
+curl -L -o $RUNNER_HOME/runner.tar.gz https://github.com/actions/runner/releases/download/v2.298.2/actions-runner-linux-x64-2.298.2.tar.gz
```

Comment:
The `Runner.Listener` program updates runner version if the installed version is not up-to-date; so I don't think I have to download latest version here.
The execution of `Runner.Listener`  is handled here as it will be terminated after the update.
https://github.com/cybozu-go/neco-gcp/blob/fc4b30bce70161e607d15c29b50cc473088a175b/cmd/necogcp/cmd/createrunner.go#L80-L84